index.d.ts

```diff
@@ -0,0 +1,456 @@
+declare module 'oidc-provider' {
+    import { Request, Response, Handler } from 'express';
+    import Koa = require('koa');
+    import Router = require('koa-router');
+    import { IncomingMessage, ServerResponse } from 'http';
+    import EventEmitter = require('events');
+    import { JWK } from 'node-jose';
+
+    type SubjectType = 'public' | 'parwise';
+
+    interface IClient {
+        applicationType?: 'web' | 'native';
+        grant_types?: string[];
+        id_token_signed_response_alg?: string;
```

Comment:
defaults to RS256

Reply:
Same issue with interfaces not being able to have default values - convert to class?

Reply:
what would be the implication of converting to class?